pkg/webui/console/views/device/index.js

```diff
@@ -74,12 +81,14 @@ import style from './device.styl'
       maySendUplink: checkFromState(maySendUplink, state),
       fetching: selectDeviceFetching(state),
       error: selectDeviceError(state),
+      applicationLinkfetching: applicationLinkfetching || !link,
```

Comment:
Just add this to the `fetching` prop on L82, the `withRequest` HOC will do the rest.

```javascript
fetching: selectDeviceFetching(state) || selectApplicationLinkFetching(state)
```

Reply:
Also avoid adding other dependencies to fetching props (like whether a link is present here). The selector should provide everything necessary to determine whether something is fetched.